code-scanning/trivy.yml

```diff
@@ -14,8 +14,13 @@ on:
   schedule:
     - cron: $cron-weekly
 
+permissions: read-all
+
 jobs:
   build:
+    permissions:
+      contents: read # for actions/checkout to fetch code
+      security-events: write # for github/codeql-action/upload-sarif to upload SARIF results
```

Comment:
Hi, thanks for the PR!

Could you explain what you're aiming for with these changes? It seems that the top-level permissions statement is superfluous given the more specific statement applied to the `build` job...?

Reply:
Hi @nickfyson, @knqyf263 these changes are part of https://github.com/actions/starter-workflows/issues/1299. @Devils-Knight and others from StepSecurity are fixing token permissions across the starter workflow templates, so they are secure by default. We have also setup a website https://app.stepsecurity.io that developers can use to set token permissions for their workflows. 

@nickfyson to answer your question, ossf/scorecards [recommends](https://github.com/ossf/scorecard/blob/main/docs/checks.md#token-permissions) adding workflow level permissions, so that if developers add a new job to the workflow, it inherits the `read-all` permissions, and is secure-by-default. 

@knqyf263 to answer your question, if job level permissions are granted, then any permission not specified is not granted to the job. So, if `contents: read` is removed, the `build` job will not have `contents-read` permission. 

CC: @laurentsimon

Reply:
Ah, thanks for the context. 🙂 

How come `read-all` though? Would it be better to just give read to contents by default and nothing else?

Reply:
@nickfyson yes, we got that feedback from multiple developers, and have updated the logic on https://app.stepsecurity.io/. This PR was created before that change though. @Devils-Knight can you please update this PR so that the workflow level permissions are set to `contents: read`?

Reply:
@varunsh-coder Thanks for explaining the changes 😇. Have updated the workflow level permission as requested by you please check it out.

Reply:
> @varunsh-coder Thanks for explaining the changes 😇. Have updated the workflow level permission as requested by you please check it out.

LGTM